### ISSUE TITLE
Feature idea: automate the installation of appliances with gns3-server installation.

### ISSUE THREAD

**Issue by AJNOURI** [GNS3/gns3-server#1436]
For work, I need to test topologies with dozens of vMX (min 8GB each) on-demand on packet.net (128 GB RAM machine) which costs a lot of money if maintained UP for long time (1.75$ per hour).

So to reduce costs, the idea is to use a script to install gns3 server + vMX applaince images. (images provided by the user)
This way, I could tear down the server when done and repeat the process for the next test.
It is a kind of CI-CD integration case  ;)

And it would be easier to convince my boss to pay on the go for big baremetal packet.net machine.

It could be a kind of appliance parameter file with pre-configured actions and path to the image files .
The file could be read right after the server is installed to install the appliance(s).

In short, eliminate user interactions and automate the process of installing appliance(s) during installation.

**Comment by grossmj** [comment]
Completing https://github.com/GNS3/gns3-server/issues/1427 is the first step to implement this.

**Comment by Paraidomat** [comment]
I also would love to see this implemented, since I have a quite similar issue. In my case I want to automate the setup, so a user just clicks a button and a fully functional GNS3 instance will be started inside a docker container.

**Comment by grossmj** [comment]
Now that we have implemented in the appliance management API in 2.2, it will be possible create a shell script to automatically add, update and remove appliances. For example, to add an IOS appliance using curl:

```
curl -X POST "http://localhost:3080/v2/appliances" -d '{"default_name_format": "R{0}", "platform": "c3725",  "name": "My c3725 router","ram": 128, "compute_id": "local", "symbol": ":/symbols/router.svg", "image": "c3725-adventerprisek9-mz.124-15.T14.image", "category": "router", "appliance_type": "dynamips"}'
```

An alternative solution would be to define the appliances in `gns3_controller.conf`.

Please wait for the alpha version to test since we still have some polishing work to do on the 2.2 branch. Thanks!

**Comment by AJNOURI** [comment]
Nice @grossmj!

**Comment by grossmj** [comment]
@AJNOURI now is the time to test this since we have release version 2.2 beta 1, use our API to add, update or remove templates: http://api.gns3.net/en/2.2/curl.html#template-creation
